<launch>
    <!--这个launch文件主要包含两个部分，一个是启动乌龟的GUI节点，第二生成一个新的乌龟-->
    <node pkg="turtlesim" type="turtlesim_node" name="turtle1" output="screen"/>
    <node pkg="turtlesim" type="turtle_teleop_key" name="keyboard" output="screen"/>
    <node pkg="turtle_exs" type="turtle_start" name="turtle2" output="screen"/>
    <!-- 基本实现思路
            节点只编写一次
            但是节点需要启动两次
            节点启动是动态传参 第一次是turtle1 第二次是turtle2 -->
    <node pkg="turtle_exs" type="turtle_tracking_pub" name="pub1" args="turtle1" output="screen"/>
    <node pkg="turtle_exs" type="turtle_tracking_pub" name="pub2" args="turtle2" output="screen"/>
    <!-- 这部分相当于就是说订阅两个小乌龟的位置转换关系 -->
    <!-- <node pkg="turtle_exs" type="turtle_tracking_sub" name="sub" output="screen"/> -->
    <node pkg="turtle_exs" type="control_turtle2" name="control_turtle2" output="screen"/>
</launch>